forum/frontend/src/components/SelectRole.js

```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { AuthContext } from '../context/AuthContext';
+import api from '../api';
+import '../styles/selectrole.css'; 
+
+const SelectRole = () => {
+  const { changeRole } = useContext(AuthContext);
+  const navigate = useNavigate();
+
+  const handleRoleChange = (role) => {
+    api.post('/api/v1/change-role/', { role }, {
+      headers: {
+        Authorization: `Bearer ${localStorage.getItem('accessToken')}`,
+      }
+    })
+    .then(response => {
+      changeRole(role);
+      navigate(`/`);
+    })
+    .catch(error => {
+      if (error.response && error.response.status === 401) {
+        console.error('Unauthorized: Invalid token or token expired');
+      } else {
+        console.error('Error changing role:', error);
```

Comment:
Users should be notified visually about errors, such as "Unauthorized" or "Role change failed," instead of just logging to the console.